sbt-metals/src/main/scala/scala/meta/metals/MetalsPlugin.scala

```diff
@@ -56,11 +56,7 @@ object MetalsPlugin extends AutoPlugin {
            } else None)).distinct
       }
     },
-    semanticdbEnabled := {
-      val old = semanticdbEnabled.value
-      if (ScalaInstance.isDotty(scalaVersion.value)) true
-      else old
```

Comment:
Am I totally missing something with this? I don't remember when I first added this why I did this. Thinking about it now, this should always just be enabled no? We do check the sbt version before we ever even get to this point, so there isn't a risk that we are enabling this in a version of sbt that doesn't support it. Also, I'm unsure why we were setting it to `true` for Dotty, but not in Scala 2. I believe this should just be enabled all the time when a user is in a valid sbt version.

Reply:
:thinking: It seems that there is some magical thing in sbt-BSP that anyway enables semanticdb for scala2 projects)
Otherwise, I have no idea how it worked before as people rarely enable this setting explicitly.

Reply:
Wasn't this due to the fact that sbt can have a wrong version of semanticdb? We are setting the compiler plugin separately, so `semanticdbEnabled` should not be needed at all in case of Scala 2. For Scala 3 it's safe as it just adds the -Xsemanticdb flag.

Reply:
ahhhhhh you're right. I totally forgot about this. 👍🏼 I'll close.

Reply:
Maybe we should add a comment though :P I also took me a while to remember this.

Reply:
👍🏼 I'm making some changes to ensure the sbt plugin gets added to the metabuild when the sbt support lands, so I'll add it in that pr.

Reply:
Thanks!